experimental/src/Config/ConfigBuilder.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Experimental\Config;
+
+use OpenTelemetry\Experimental\Config\Exporter\Otlp\Config as OtlpConfig;
+use OpenTelemetry\Experimental\Config\SpanProcessor\BatchSpanProcessorConfig;
+use OpenTelemetry\Experimental\Config\SpanProcessor\NoopSpanProcessorConfig;
+use OpenTelemetry\Experimental\Config\SpanProcessor\SimpleSpanProcessorConfig;
+
+class ConfigBuilder
+{
+    //initial list of known exporter types (exluding exotic contrib exporters, which must be added)
+    private array $exporters = [
+        OtlpConfig::class,
+    ];
+    //initial list of known span processor types
+    private array $spanProcessors = [
+        BatchSpanProcessorConfig::class,
+        SimpleSpanProcessorConfig::class,
+        NoopSpanProcessorConfig::class,
+    ];
+    private array $userConfig = [];
+    private array $environmentConfig = [];
+
+    public function __construct()
+    {
+        $this->environmentConfig = $this->getEnvironmentConfig();
+    }
+
+    public function build(): object
+    {
+        $config = new Config($this->userConfig, $this->environmentConfig);
+        foreach ($this->buildExporters() as $name => $exporterConfig) {
+            $config->addExporter($name, $exporterConfig);
+        }
+        foreach ($this->buildSpanProcessors() as $name => $spanProcessorConfig) {
+            $config->addSpanProcessor($name, $spanProcessorConfig);
+        }
+
+        return $config;
+    }
+
+    public function withUserConfig(array $config): self
+    {
+        $this->userConfig = $config;
+
+        return $this;
+    }
+
+    public function withExporterConfig(string $klass): self
+    {
+        $this->exporters[] = $klass;
+
+        return $this;
+    }
+
+    /**
+     * Retrieve all non-empty OTEL_* environment variables
+     */
+    private function getEnvironmentConfig(): array
+    {
+        return array_filter(getenv(), function ($value, $key) {
```

Comment:
This is like "Select * from database". 
We just cannot make assumptions on how many env vars there are, and this seem like it has to be done with every request again. This could be easily solved, if the env vars were constants.